app/src/main/java/com/amaze/filemanager/adapters/RecyclerAdapter.java

```diff
@@ -540,9 +540,9 @@ public void onBindViewHolder(final RecyclerView.ViewHolder vholder, int p) {
                 });
 
                 // resetting icons visibility
-                holder.genericIcon.setVisibility(View.GONE);
-                holder.pictureIcon.setVisibility(View.GONE);
-                holder.apkIcon.setVisibility(View.GONE);
+                holder.genericIcon.setVisibility(View.VISIBLE);
+                holder.pictureIcon.setVisibility(View.INVISIBLE);
+                holder.apkIcon.setVisibility(View.INVISIBLE);
```

Comment:
These seem to be in the same `FrameLayout` (in list at least) which seems that would make problematic having views invisible but still taking space.

Reply:
This is the best solution I came up with. We can always debug the whole layout but I didn't have the time.